packager/react-native-xcode.sh

```diff
@@ -23,11 +23,23 @@ case "$CONFIGURATION" in
     ;;
 esac
 
-# Xcode project file for React Native apps is located in ios/ subfolder
-cd ..
+search_parent_paths_for() {
+  if [ -e "$1" ]; then
+    pwd
+    return 0
+  elif [ "$PWD" != "/" ]; then
+    (cd ..; search_parent_paths_for "$1")
+    return $?
+  fi
+  return 1
+}
 
-set -x
-DEST=$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH
+# Load user environment (which should also handle ndenv or nvm setup)
+[ -e "$HOME/.bashrc" ] && source "$HOME/.bashrc"
+
+# Add node binstubs to our binary search $PATH
+project_root_path="$(search_parent_paths_for node_modules)"
+export PATH="$PATH:$project_root_path/node_modules/.bin"
```

Comment:
I don't think this assumption is (or will always be) valid. npm3 changes the way it structures node_modules, individuals can use `npm link` to work on react-native, etc. etc. Using heuristics like traversing file system caused a lot of troubles for us internally, so I would strongly advice against it.

Also the `react-native` command you get from `node_modules/.bin` is not the `react-native` command user installed globally with `npm install -g react-native-cli`. It's actually a stub that is used to warn user that they installed `react-native` instead of `react-native-cli`.